components/tikv_util/src/codec/stream_event.rs

```diff
@@ -42,14 +62,45 @@ impl EventIterator<'_> {
         self.offset += 4;
         result
     }
+
+    fn move_to_next_key_without_rewrite(&mut self) {
+        let key_len = self.get_size() as usize;
+        self.key_buf.clear();
+        self.key_buf.reserve(key_len);
+        self.key_buf
+            .extend_from_slice(&self.buf[self.offset..self.offset + key_len]);
+        self.offset += key_len;
+    }
+
+    fn move_to_next_key_with_rewrite(&mut self) {
+        let key_len = self.get_size() as usize;
+        let rewrite = self.rewrite_rule.as_ref().expect("rewrite rule not set");
+        if key_len < rewrite.from.len()
+            || &self.buf[self.offset..self.offset + rewrite.from.len()] != rewrite.from
+        {
+            self.move_to_next_key_without_rewrite();
```

Comment:
if runs to here, there must be some wrong. how about print the log?

Reply:
I think we can check this outside🤔